driver-sync/src/test/functional/com/mongodb/client/model/expressions/InContextExpressionsFunctionalTest.java

```diff
@@ -0,0 +1,160 @@
+        col.drop();
+    }
+
+    @AfterEach
+    public void tearDown() {
+        client.close();
+    }
+
+    private String bsonToString(final Bson project) {
+        return project.toBsonDocument(Document.class, MongoClientSettings.getDefaultCodecRegistry()).toString().replaceAll("\"", "'");
+    }
+
+    private List<Document> aggregate(final Bson... stages) {
+        AggregateIterable<Document> result = col.aggregate(Arrays.asList(stages));
+        List<Document> results = new ArrayList<>();
+        result.forEach(r -> results.add(r));
+        return results;
+    }
+
+    @Test
+    public void matchTest() {
+        col.insertMany(Arrays.asList(
+                Document.parse("{_id: 1, x: 0, y: 2}"),
+                Document.parse("{_id: 2, x: 0, y: 3}"),
+                Document.parse("{_id: 3, x: 1, y: 3}")));
+
+        List<Document> results = aggregate(
+                match(Filters.expr(current().getInteger("x").eq(of(1)))));
+
+        assertEquals(
+                Arrays.asList(Document.parse("{_id: 3, x: 1, y: 3}")),
+                results);
+    }
+
+    @Test
+    public void projectTest() {
+        col.insertMany(Arrays.asList(
+                Document.parse("{_id: 1, x: 0, y: 2}")));
+
+        List<Document> expected = Arrays.asList(Document.parse("{_id: 1, x: 0, c: 2}"));
+
+        // old, using "$y"
+        Bson projectOld = project(fields(include("x"), computed("c",
+                "$y")));
+        assertEquals("{'$project': {'x': 1, 'c': '$y'}}", bsonToString(projectOld));
+        assertEquals(expected,
+                aggregate(projectOld));
+
+        // new, using current() with add/subtract
+        Bson projectNew = project(fields(include("x"), computed("c",
+                current().getInteger("y").add(10).subtract(10))));
+        assertEquals(
+                "{'$project': {'x': 1, 'c': "
+                        + "{'$subtract': [{'$add': [{'$getField': "
+                        + "{'input': '$$CURRENT', 'field': 'y'}}, 10]}, 10]}}}",
+                bsonToString(projectNew));
+        assertEquals(expected,
+                aggregate(projectNew));
+    }
+
+    @Test
+    public void projectTest2() {
+        col.insertMany(Arrays.asList(Document.parse("{_id: 0, x: 1}")));
+
+        // new, nestedArray
+        Bson projectNestedArray = project(fields(excludeId(), computed("nestedArray", ofArray(
+                current().getInteger("x").max(of(4)),
+                current().getInteger("x"),
+                of(0), of(1), of(true), of(false)
+        ))));
+        assertEquals(
+                Arrays.asList(Document.parse("{ nestedArray: [ 4, 1, 0, 1, true, false ] }")),
+                aggregate(projectNestedArray));
+
+        // new, document
+        Bson projectDocument = project(fields(computed("nested",
+                //BsonDocument.parse("{ x: {$max : ['$x', 4] }}")
+                of(new Document()).setField("x", current().getInteger("x").max(of(4)))
```

Comment:
I'm not sure what to think of empty documents - it seems that we could leave them out for now. I don't think the example in the test needs to be particularly realistic, but I updated it to be a non-empty document. (I did not use ofEmptyMap because I don't expect that it applies anymore, due to non-inheritance).

Reply:
Yes, if we are not doing `of(new Document())`, it appears to be fine to not introduce `DocumentExpression of()`.